omnibus/python-scripts/packages.py

```diff
@@ -0,0 +1,165 @@
+        print(f"Error: {e.stderr}")
+        return e.stdout, e.stderr
+
+def extract_version(specifier):
+    """
+    Extract version from the specifier string.
+    """
+    try:
+        # Get the first version specifier from the specifier string
+        return str(next(iter(pkg_resources.Requirement.parse(f'{specifier}').specifier)))
+    except Exception:
+        return None
+
+def prerm_python_installed_packages_file(directory):
+    """
+    Create prerm installed packages file path.
+    """
+    return os.path.join(directory, '.prerm_python_installed_packages.txt')
+
+def postinst_python_installed_packages_file(directory):
+    """
+    Create postinst installed packages file path.
+    """
+    return os.path.join(directory, '.postinst_python_installed_packages.txt')
+
+def diff_python_installed_packages_file(directory):
+    """
+    Create diff installed packages file path.
+    """
+    return os.path.join(directory, '.diff_python_installed_packages.txt')
+
+def requirements_agent_release_file(directory):
+    """
+    Create requirements agent release file path.
+    """
+    return os.path.join(directory, 'requirements-agent-release.txt')
+
+def create_python_installed_packages_file(filename):
+    """
+    Create a file listing the currently installed Python dependencies.
+    """
+    print(f"Creating file: '{filename}'")
+    with open(filename, 'w', encoding='utf-8') as f:
+        f.write(DO_NOT_REMOVE_WARNING_HEADER)
+        installed_packages = importlib.metadata.distributions()
+        for dist in installed_packages:
+            f.write(f"{dist.metadata['Name']}=={dist.version}\n")
+    os.chown(filename, pwd.getpwnam('dd-agent').pw_uid, grp.getgrnam('dd-agent').gr_gid)
+
+def create_diff_installed_packages_file(directory, old_file, new_file):
+    """
+    Create a file listing the new or upgraded Python dependencies.
+    """
+    old_packages = load_requirements(old_file)
+    new_packages = load_requirements(new_file)
+    diff_file = diff_python_installed_packages_file(directory)
+    print(f"Creating file: '{diff_file}'")
+    with open(diff_file, 'w', encoding='utf-8') as f:
+        f.write(DO_NOT_REMOVE_WARNING_HEADER)
+        for package_name, (_, new_req_value) in new_packages.items():
+            old_req = old_packages.get(package_name)
+            if old_req:
+                _, old_req_value = old_req
+                # Extract and compare versions
+                old_version_str = extract_version(str(old_req_value.specifier))
+                new_version_str = extract_version(str(new_req_value.specifier))
+                if old_version_str and new_version_str:
+                    if version.parse(new_version_str) > version.parse(old_version_str):
+                        f.write(f"{new_req_value}\n")
+            else:
+                # Package is new in the new file; include it
+                f.write(f"{new_req_value}\n")
+
```

Comment:
I suppose this could also be `chown`ed to `dd-agent` for consistency with files created by `create_python_installed_packages_file`

Reply:
Fixed! Good catch!